File: src/etch/bindings/java/util/test/TestStrIntHashMapSerializer.java
/* $Id$
 *
 * Created by wert on Jun 25, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.util.test;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import etch.bindings.java.msg.StructValue;
import etch.bindings.java.msg.Type;
import etch.bindings.java.support.DefaultValueFactory;
import etch.bindings.java.util.StrIntHashMap;
import etch.bindings.java.util.StrIntHashMapSerializer;

public class TestStrIntHashMapSerializer
{
	private final DefaultValueFactory vf = new DefaultValueFactory();
	
	@Test
	public void testImport0() throws Exception
	{
		StrIntHashMap map = new StrIntHashMap();
		testMap( map );
	}
	
	@Test
	public void testImport1() throws Exception
	{
		StrIntHashMap map = new StrIntHashMap();
		map.put( "a", 1 );
		testMap( map );
	}
	
	@Test
	public void testImport2() throws Exception
	{
		StrIntHashMap map = new StrIntHashMap();
		map.put( "a", 1 );
		map.put( "b", 2 );
		testMap( map );
	}
	
	@Test
	public void testImport3() throws Exception
	{
		StrIntHashMap map = new StrIntHashMap();
		map.put( "a", 1 );
		map.put( "b", 2 );
		map.put( "c", 3 );
		testMap( map );
	}
	
	@Test
	public void testImportNull1() throws Exception
	{
		StrIntHashMap map = new StrIntHashMap();
		map.put( "a", null );
		testMap( map );
	}
	
	@Test
	public void testImportNull2() throws Exception
	{
		StrIntHashMap map = new StrIntHashMap();
		map.put( null, 2 );
		testMap( map );
	}
	
	@Test
	public void testImportNull3() throws Exception
	{
		StrIntHashMap map = new StrIntHashMap();
		map.put( null, null );
		testMap( map );
	}
	
	public void testMap( StrIntHashMap map )
	{
		Type type = vf.getType( "blah" );
		StrIntHashMapSerializer.init( vf, type );
		
		StructValue sv = new StructValue( type );
		StrIntHashMapSerializer.exportValue( vf, map, sv );
//		System.out.println( "sv = "+sv );
		StrIntHashMap map2 = StrIntHashMapSerializer.importValue( vf, sv );
		assertEquals( map, map2 );
	}
}

File: src/etch/bindings/csharp/util/test/TestStrIntHashMapSerializer.cs
using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;
using Etch.Util;
using Etch.Support;
using Etch.Msg;
namespace Etch.Util.Test
{
    [TestFixture]
    public class TestStrIntHashMapSerializer
    {

        [TestFixtureSetUp]
        public void First()
        {
            Console.WriteLine();
            Console.Write( "TestStrIntHashMapSerializer" );
        }


        [Test]
        public void TestImport0() 
	    {
		    StrIntHashMap map = new StrIntHashMap();
		    TestMap( map );
	    }

        [Test]
	    public void TestImport1() 
	    {
		    StrIntHashMap map = new StrIntHashMap();
		    map.Add( "a", 1 );
		    TestMap( map );
	    }

        [Test]
	    public void TestImport2() 
	    {
		    StrIntHashMap map = new StrIntHashMap();
		    map.Add( "a", 1 );
		    map.Add( "b", 2 );
		    TestMap( map );
	    }

        [Test]
	    public void TestImport3() 
	    {
		    StrIntHashMap map = new StrIntHashMap();
		    map.Add( "a", 1 );
		    map.Add( "b", 2 );
		    map.Add( "c", 3 );
		    TestMap( map );
	    }
    	
        [Test]
	    public void TestImportNull1() 
	    {
		    StrIntHashMap map = new StrIntHashMap();
            map.Add( "a", 1 );
		    map.Add( "b", null );
            map.Add( "c", 2 );
		    TestMap( map );
	    }
    	
	    // no tests for null keys as string? isn't allowed

        public void TestMap( StrIntHashMap map )
        {
            XType type = new XType( "map" );
		  
            Class2TypeMap class2type = new Class2TypeMap();
		    StrIntHashMapSerializer.Init( type, class2type );
		    ImportExportHelper helper = type.GetImportExportHelper();
    		
		    StructValue sv = helper.ExportValue( map );
    		
		    Assert.AreEqual( sv.GetXType, type );
    		
		    StrIntHashMap map2 = (StrIntHashMap) helper.ImportValue( sv );
    		
		    Assert.AreEqual( map, map2 );
        }

    }
}
